main.go

```diff
@@ -85,46 +86,88 @@ func convertJunitToSlack(junitFiles ...*junit.Suites) []slack.Attachment {
 				titleSectionBlock := slack.NewSectionBlock(titleTextBlock, nil, nil)
 				failedTestsBlocks = append(failedTestsBlocks, titleSectionBlock)
 
-				failureMessage := result.Failure.Message
+				// Slack has a 3000-character limit for (non-field) text objects
+				failureMessage := failure.Message
+				if len(failureMessage) > 3000 {
+					failureMessage = failureMessage[:3000]
+				}
 
-				// If there's no failure message, we'll use a different message (this shouldn't be the usual case)
+				// Slack has a 3000-character limit for (non-field) text objects
+				failureValue := failure.Value
+				if len(failureValue) > 3000 {
+					failureValue = failureValue[:3000]
+				}
+
+				// Add some formatting to the failure title
+				failureTitleTextBlock := slack.NewTextBlockObject("plain_text", title, false, false)
+				failureTitleHeaderBlock := slack.NewHeaderBlock(failureTitleTextBlock)
+
+				// If there's no failure message or value, use a different message (this shouldn't be the usual case)
 				if failureMessage == "" {
```

Comment:
I think I would find this easier to read if it was:
```
If failureMessage == "" && failureValue == "" {
    handleNoFailureDetails()
}
if failureMessage != "" {
    addAttachmentFromFailureMessage()
}
if failureValue != "" {
    addAttachmentFromFailureValue()
}
```
Maybe addAttachmentFromFailureValue() addAttachmentFromFailureMessage() could be the same func.

Reply:
Nah, it's not you. I wasn't a huge fan of it while writing it. You pointing it out proves it